desktop/core/src/desktop/middleware.py

```diff
@@ -130,11 +130,9 @@ def process_view(self, request, view_func, view_args, view_kwargs):
     if "fs" in view_kwargs:
       del view_kwargs["fs"]
 
-    request.fs = None
+    request.fs = fsmanager.get_filesystem(request.fs_ref)
```

Comment:
Interesting, I guess it failed because https://github.com/cloudera/hue/blob/master/desktop/core/src/desktop/auth/views_test.py#L68 is not authenticated. Would need to double check why, as here we would give a FS to a non authenticated user which could be a security issue.

Would it make more sense to instead get the fs in https://github.com/cloudera/hue/blob/master/desktop/core/src/desktop/auth/views.py#L149 if it is None? (so that we are 100% sure the user is authenticated)

Reply:
As you said, the client isn't authenticated.

But, with this backend `desktop.auth.backend.AllowFirstUserDjangoBackend` , 
the first user `test_username`  would be allowed and be an admin. (so, the test succeed )

With `desktop.auth.backend.AllowFirstUserDjangoBackend` ,
I think middleware can't(don't have to) be sure the user is authenticated.

And, before using fs in the `ensure_home_directory`, it checks userprofile. 
For this case, it doesn't have a security issue.
https://github.com/cloudera/hue/blob/d3cc8ff366620b90670121eea0bad013cfd74dc1/apps/useradmin/src/useradmin/views.py#L854